pkg/disk/filesystem.go

```diff
@@ -22,6 +22,8 @@ type Filesystem struct {
 	FSTabFreq uint64 `json:"fstab_freq,omitempty" yaml:"fstab_freq,omitempty"`
 	// The sixth field of fstab(5); fs_passno
 	FSTabPassNo uint64 `json:"fstab_passno,omitempty" yaml:"fstab_passno,omitempty"`
+	// Custom mkfs options
+	MkfsOptions []string `json:"mkfs_options,omitempty" yaml:"mkfs_options,omitempty"`
```

Comment:
I'm a little bit uncertain about this, it feels a bit loose, i.e. a mistype/misremember to e.g. `verify`would not have the desired effect in the backend but also leads to no error (also no validation that verity will only work on ext4).

Reply:
Yes. I wanted this to be a bit flexible for later extensions, but maybe we can tighten the golang side of things while leaving the json side flexible for extensions? Like, by adding some enum?

Reply:
I haven't thought this through entirely, but I think for the disk types this is probably ok.  I consider it a low level interface and it'll be hard to cover every available mkfs option here.  But at a higher level, for the manifest generator that calls into the partition table generator and wants to set options, we can be better about this.

Reply:
Yeah, I like this! Lets keep the json flexible and just the golang API strict(er), if you have the time to open something that would be great (if not that is fine too, just let us know).

Reply:
I'll have a look at that.

Reply:
See https://github.com/osbuild/images/pull/1526